easybuild/tools/systemtools.py

```diff
@@ -410,6 +410,26 @@ def get_tool_version(tool, version_option='--version'):
         return '; '.join(out.split('\n'))
 
 
+def get_gcc_version():
+    """
+    Process `gcc --version` and return the GCC version.
+    """
+    out, ec = run_cmd('gcc --version', simple=False, log_ok=False, force_in_dry_run=True)
+    if ec:
+        _log.warning("Failed to determine the version of GCC: %s" % out)
+        return UNKNOWN
+
+    find_version = re.search("^gcc\s+\([^)]+\)\s+(?P<version>[^\s]+)\s+", out)
+    if find_version:
+        return find_version.group('version')
```

Comment:
don't return inline (use `res = find_version_group('version')`, and `return res` at the end), add a log debug mentioning both `out` and the extracted version